deepcell/model_zoo/panopticnet.py

```diff
@@ -355,35 +300,36 @@ def PanopticNet(backbone,
             else:
                 input_shape_with_time = tuple(
                     [frames_per_batch] + list(input_shape))
-            inputs = Input(shape=input_shape_with_time)
+            inputs = Input(shape=input_shape_with_time, name='input_0')
         else:
-            inputs = Input(shape=input_shape)
+            inputs = Input(shape=input_shape, name='input_0')
 
-    # force the channel size for backbone input to be `required_channels`
+    # Normalize input images
     if norm_method is None:
         norm = inputs
     else:
         if frames_per_batch > 1:
-            norm = TimeDistributed(ImageNormalization2D(norm_method=norm_method))(inputs)
+            norm = TimeDistributed(ImageNormalization2D(
+                norm_method=norm_method, name='norm'))(inputs)
         else:
-            norm = ImageNormalization2D(norm_method=norm_method)(inputs)
+            norm = ImageNormalization2D(norm_method=norm_method, name='norm')(inputs)
 
+    # Add location layer
     if location:
         if frames_per_batch > 1:
             # TODO: TimeDistributed is incompatible with channels_first
-            loc = TimeDistributed(Location2D(in_shape=input_shape))(norm)
+            loc = TimeDistributed(Location2D(in_shape=input_shape, name='location'))(norm)
         else:
-            loc = Location2D(in_shape=input_shape)(norm)
-        concat = Concatenate(axis=channel_axis)([norm, loc])
+            loc = Location2D(in_shape=input_shape, name='location')(norm)
+        concat = Concatenate(axis=channel_axis, name='concatenate_location')([norm, loc])
     else:
         concat = norm
 
-    if frames_per_batch > 1:
-        fixed_inputs = TimeDistributed(TensorProduct(required_channels))(concat)
-    else:
-        fixed_inputs = TensorProduct(required_channels)(concat)
+    # Force the channel size for backbone input to be `required_channels`
+    fixed_inputs = conv(required_channels, conv_kernel, strides=1,
+                        padding='same', name='conv_channels')(concat)

```

Comment:
Is this the same thing? I thought that `TensorProduct` is more like a `Dense` layer.

Reply:
I am mistaken.

`TensorProduct` is the mathematical equivalent to a `Conv` with a 1x1 filter, which is why we instantiate `conv_kernel = (1, 1)` (for 2D).  `Dense` layers, on the other hand, flatten the inputs, which is not desired in this case.